api/routes/units.js

```diff
@@ -0,0 +1,91 @@
+const router = require('express').Router()
+const Units = require('../models/units')
+const Company = require('../models/company')
+
+
+//Get all routes and queries. 
+router.get('/units', async (req, res, next) => {
+    const status = 200
+
+     if(typeof req.query.kind !== 'undefined'){
+         await Units.find().select('kind').then(response => {
+            res.json({ status, response })
+         })
+
+    } else if (typeof req.query.floor !== 'undefined') {
+        await Units.find().select('floor').then(response => {
+            res.json({ status, response })
+         })
+
+    } else if (typeof req.query.occupied !== 'undefined') {
+        await Units.find({company: []}, (err, result) => {
+            if (err) {console.log(err)}
+            if (!result.length) {
+                 Units.find({company: []}).then(response => {
```

Comment:
I don't think you should need to re-request this information. You should just be able to respond with `result`.